readthedocs/proxito/views/serve.py

```diff
@@ -673,3 +675,45 @@ def changefreqs_generator():
 
 class ServeSitemapXML(SettingsOverrideObject):
     _default_class = ServeSitemapXMLBase
+
+
+class ServeStaticFiles(CachedView, CacheTagsMixin, View):
+
+    """
+    Serve static files from the same domain the docs are being served from.
+
+    This is basically a proxy for ``STATIC_URL``.
+    """
+
+    project_cache_tag = "staticfile"
+    cache_request = True
+
+    @method_decorator(map_project_slug)
+    def get(self, request, filename, project):
+        try:
+            # This is needed for the _get_project
+            # method for the CacheTagsMixin class.
+            self.project = project
+            return FileResponse(staticfiles_storage.open(filename))
+        except FileNotFoundError:
+            # Don't use ``raise Http404`` so the other operations
+            # in ``.dispatch`` can be completed.
+            return HttpResponse(status=404)
```

Comment:
We could also just don't set the cache headers for 404s. Also, what a bummer there isn't a way to run some logic in all cases for a specific view without using a middleware

Reply:
> We could also just don't set the cache headers for 404s.

I imagine we shouldn't have any 404's on these URL's, but bots have a way of generating junk URL's really effectively.

> Also, what a bummer there isn't a way to run some logic in all cases for a specific view without using a middleware

You mean like a global exception handler for the view or something?

Reply:
> You mean like a global exception handler for the view or something?

yeah